Compile error with MSVC

Godot 3.0 fails to compile with MSVC because of its rules for searching include files. Specifically, when compiling __`modules\gdnative\godot\dictionary.cpp`__ you get this error:

> [...]\core/io/json.h(62): error C2079: 'JSON::Token::value' uses undefined class 'Variant'

The problem is that although you'd expect `#include "variant.h"` in __`core/io/json.h`__ to include __`core/variant.h`__, since MSVC doesn't find it right where __`json.h`__ is, it looks for it according to this:

> In the directories of the currently opened include files, in the reverse order in which they were opened. The search begins in the directory of the parent include file and continues upward through the directories of any grandparent include files.

That is, it seems to go up in the include chain until it reaches the directory of the main file being compiled (__`modules/gdnative/godot`__), which happens to contain a __`variant.h`__ file itself, and that is taken.

I'm not sure how to deal with this. Happens at least with MSVC 2017.

@vnen, have you seen this issue? Any ideas on how to fix?

--- a/modules/gdnative/godot/dictionary.cpp
+++ b/modules/gdnative/godot/dictionary.cpp
@@ -29,9 +29,10 @@
 /*************************************************************************/
 #include <godot/dictionary.h>
 
+#include "core/variant.h"
+
 #include "core/dictionary.h"
 #include "core/io/json.h"
-#include "core/variant.h"
 
 #ifdef __cplusplus
 extern "C" {

